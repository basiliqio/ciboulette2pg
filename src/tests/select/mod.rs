use super::*;

mod query_params;
mod select_success;

async fn test_select<'a>(
    transaction: &mut sqlx::Transaction<'_, sqlx::Postgres>,
    query_end: &str,
    _name: &str,
    _data: &BTreeMap<String, Vec<Uuid>>,
) -> Vec<sqlx::postgres::PgRow> {
    let ciboulette_store = gen_bag();
    let table_store = gen_table_store(&ciboulette_store);
    let parsed_url = Url::parse(format!("http://localhost{}", query_end).as_str()).unwrap();
    const INTENTION: CibouletteIntention = CibouletteIntention::Read;

    let req_builder = CibouletteInboundRequestBuilder::new(INTENTION, &parsed_url, &None);
    let request = req_builder.build(&ciboulette_store).unwrap();
    let ciboulette_request = CibouletteReadRequest::try_from(request).unwrap();
    let builder = Ciboulette2PostgresBuilder::gen_select(
        &ciboulette_store,
        &table_store,
        &ciboulette_request,
    )
    .unwrap();
    let (query, args) = builder.build().unwrap();
    println!(
        "{}",
        sqlformat::format(
            query.as_str(),
            &sqlformat::QueryParams::None,
            sqlformat::FormatOptions::default()
        )
    );
    let raw_rows: Vec<sqlx::postgres::PgRow> = sqlx::query_with(&query, args)
        .fetch_all(&mut *transaction)
        .await
        .unwrap();
    raw_rows
}
