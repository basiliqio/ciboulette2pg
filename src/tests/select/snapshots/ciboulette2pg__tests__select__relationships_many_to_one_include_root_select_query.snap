---
source: src/tests/select/mod.rs
expression: "sqlformat::format(query.as_str(), &sqlformat::QueryParams::None,\n                  sqlformat::FormatOptions::default())"

---
WITH "cte_peoples_data_0" AS (
  SELECT
    "public"."peoples"."id" :: TEXT AS "id",
    "public"."peoples"."id" AS "main_id",
    "public"."comments"."id" :: TEXT AS "related_id",
    $1 :: TEXT AS "related_type",
    $2 :: TEXT AS "type",
    NULL :: json AS "data",
    "public"."peoples"."favorite_color" AS "rel_favorite_color"
  FROM
    "public"."peoples"
    INNER JOIN "public"."comments" ON "public"."comments"."author" = "public"."peoples"."id"
  WHERE
    "public"."comments"."id" = $3
),
"cte_rel_comments_data_1" AS (
  SELECT
    "public"."comments"."id" :: TEXT AS "id",
    "public"."comments"."id" AS "main_id",
    NULL :: TEXT AS "related_id",
    NULL :: TEXT AS "related_type",
    $4 :: TEXT AS "type",
    NULL :: json AS "data",
    "public"."comments"."article" AS "rel_article",
    "public"."comments"."author" AS "rel_author"
  FROM
    "public"."comments"
  WHERE
    "public"."comments"."id" = $5
),
"cte_rel_articles_data_2" AS (
  SELECT
    "public"."articles"."id" :: TEXT AS "id",
    "public"."articles"."id" AS "main_id",
    "cte_peoples_data_0"."id" :: TEXT AS "related_id",
    $6 :: TEXT AS "related_type",
    $7 :: TEXT AS "type",
    NULL :: json AS "data"
  FROM
    "public"."articles"
    INNER JOIN "public"."people-article" ON "public"."people-article"."article_id" = "public"."articles"."id"
    INNER JOIN "cte_peoples_data_0" ON "cte_peoples_data_0"."main_id" = "public"."people-article"."people_id"
) (
  SELECT
    "cte_peoples_data_0"."id",
    "cte_peoples_data_0"."type",
    "cte_peoples_data_0"."data",
    "cte_peoples_data_0"."related_type",
    "cte_peoples_data_0"."related_id"
  FROM
    "cte_peoples_data_0"
);
